24_clustering.Rmd

```diff
@@ -221,97 +196,122 @@ tse <- GlobalPatterns
 
 ```{r dmm6}
-# Get the model that has the best fit
-bestFit <- metadata(tse)$DMM$dmm[[metadata(tse)$DMM$best]]
-bestFit
+head(colData(tse_dmm)$clusters, 10)
 ```
+More detailed information about the clusters can be accessed in the metadata. The metadata contains samples-cluster assignment probabilities that tell us the likelihood for each sample to belong to each cluster.
 
-### PCoA for ASV-level data with Bray-Curtis; with DMM clusters shown with colors
+```{r dmm7}
+head(metadata(tse_dmm)$DMM$prob, 10) 
+# Can be accessed from in best model infromation as well using:
+# metadata(tse_dmm)$DMM$dmm[2][[1]]@group
+```
 
-Group samples and return DMNGroup object that contains a summary.
-Patient status is used for grouping.
 
-```{r dmm_group}
-dmm_group <- calculateDMNgroup(tse, variable = "SampleType", 
-                               assay.type = "counts", k = 2, 
-                               seed = .Machine$integer.max)
+Once the optimal model have been confirmed, we can find out which samples are grouped with each other. The table below shows one sample of each sample type clustered in either of the groups. We can notice that DMM can distinguish environmental samples into one group, and mock and human samples into another. For clarity, in this example, the probabilities for each sample to belong in each cluster have been rounded. 
 
-dmm_group
-```
+```{r dmm8}
+library(dplyr)
 
-Mixture weights (rough measure of the cluster size).
+clusters <- round(metadata(tse_dmm)$DMM$prob, 1)
+clusters <- as.data.frame(cbind(clusters, levels(colData(tse_dmm)$SampleType)[colData(tse_dmm)$SampleType])) # add sample type information
+colnames(clusters) <- c("Group1", "Group2", "SampleType")
 
-```{r dmm7}
-DirichletMultinomial::mixturewt(bestFit)
+clusters %>%
+  group_by(SampleType) %>%
+  arrange(Group1) %>%
+  filter(row_number()==1)
 ```
 
-It's also possible to get the samples-cluster assignment probabilities: how
-probable it is that each sample belongs to each cluster
+We can also plot the driver Phyla in each group. In this case, it reflects the differences between environmental and human samples. 
 
-```{r dmm8}
-prob <- metadata(tse)$DMM$prob
-head(prob)
+```{r dmm9, fig.width=8, fig.height=4}
+# Get the estimates on how much each phyla contributes on each cluster
+best_model <- metadata(tse_dmm)$DMM$dmm[2]
+drivers <- as.data.frame(best_model[[1]]@fit$Estimate)
+
+drivers$phyla <- gsub("Phylum:", "", rownames(drivers)) # Clean phylum names
+
+plots <- c()
+for (i in 1:2) {
+  drivers <- drivers[order(drivers[[i]], decreasing = TRUE),]
+  p <- ggplot(head(drivers, 10), aes(x = reorder(head(phyla, 10), + head(drivers[[i]], 10)), y = head(drivers[[i]], 10))) +
+          geom_bar(stat = "identity", fill = "deeppink4", alpha = 0.5) +
+          coord_flip() + labs(title = paste("Top phyla in a group", i)) +
+          theme_light(base_size = 15) + xlab("") + ylab("") + ylim(0, 7)
```

Comment:
I think xlab & ylab and ylim are not any more recommended; use instead:

labs(x="", y="") and scale_y_continuous(limits=c(0,7))